### PR TITLE
Add stitch tolerance to satin underlays

### PR DESCRIPTION
**Pull request by kaalleen**
Add stitch tolerance option to contour and center walk underlay.
Especially small fonts do have issues with the underlay showing through.
The tolerance defaults to the respective stitch length setting.

### CODE REVIEW

**Review by lexelby — APPROVED** [review]
Love it, nice work!

File: lib/elements/satin_column.py [hunk 1]
```diff
@@ -359,6 +359,24 @@ def contour_underlay(self):
...
+            'A higher tolerance means sharp corners may be rounded. '
+            'Defaults to stitch length.'
+        ),
+        unit='mm',
+        group=_('Contour Underlay'),
+        type='float',
+    )
+    def contour_underlay_stitch_tolerance(self):
+        tolerance = self.get_float_param("contour_underlay_stitch_tolerance_mm", self.contour_underlay_stitch_length)
+        return max(tolerance, 0.01)
```
Comment (george-steel) [inline lib/elements/satin_column.py:378]: This currently chokes on unset tolerances due to the implicit mm to px conversion of `get_float_param` getting applied twice, breaking underlay on all older files. I would suggest just giving tolerance a default of 0.2mm as this gives sensible behaviour for underlay.

Comment (kaalleen) [reply lib/elements/satin_column.py:378]: Thanks for the catch. Wow, I wasn't aware we were inconsistent on returning param values. While we return the default value directly for int values, we do mm calculations on floats. I don't think that's good. We should return the defaults directly as they are given without any further changes.

Comment (george-steel) [reply lib/elements/satin_column.py:378]: If we're going to do a large refactor making that large a change (as currently all of the hardcoded defaults are in mm), we should just split off `get_mm_param_as_px` into its own function and get rid of the magic param names.

Comment (george-steel) [reply lib/elements/satin_column.py:378]: I added a fix to my current PR #2830

Comment (george-steel) [reply lib/elements/satin_column.py:378]: Unfortunately, the running stitch algorithm tends to blow up when tolerances are that large, especially when its input is also such a coarse sample. Using the stitch length as the tolerance worked on exactly none of the curved satin columns I tested it on, all of which worked with the old underlay.

My proposed default value of 0.2mm is not chosen arbitrarily. It is chosen as half of the default inset, which is value needed by the running stitch and sampling algorithm to "just work" and reliably stay underneath the satin column.

Comment (george-steel) [reply lib/elements/satin_column.py:378]: Basically, the 0.2 default is for backwards compatibility. Larger tolerances might be useful in some cases, but applying them as a default will break existing designs.

**Pull request merged**